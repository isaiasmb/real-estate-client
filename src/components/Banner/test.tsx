import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Banner from '.'

const props = {
  img: 'https://source.unsplash.com/user/fromitaly/1042x580',
  title: 'Nemo enim ipsam',
  subtitle: '<p>Ut labore et <strong>dolore magnam</strong> aliquam',
  buttonLabel: 'Buy now',
  buttonLink: '/properties/nemo-enim-ipsam'
}

describe('<Banner />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Banner {...props} />)

    expect(
      screen.getByRole('heading', { name: /Nemo enim ipsam/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', {
        name: /Ut labore et dolore magnam aliquam/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: /Nemo enim ipsam/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
