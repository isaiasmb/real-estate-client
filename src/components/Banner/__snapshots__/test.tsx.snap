// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctly 1`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #234DD4;
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.2rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c6:hover {
  background: #1c3da8;
}

.c3 {
  max-width: 130rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(3.2rem / 2);
  padding-right: calc(3.2rem / 2);
}

.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 23rem;
  background-color: #EAEAEA;
  background-image: url(https://source.unsplash.com/user/fromitaly/1042x580);
  background-position: center center;
  background-size: cover;
}

.c2 {
  width: 100%;
  background-color: rgba(0,0,0,0.7);
  padding: 2.4rem 0;
  padding-bottom: 4.0rem;
}

.c4 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c5 {
  color: #FAFAFA;
  font-size: 1.4rem;
  font-weight: 400;
  margin-bottom: 1.6rem;
}

.c5 strong {
  color: #234DD4;
  font-weight: 600;
}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
  }
}

@media (min-width:768px) {
  .c1 {
    height: 48rem;
  }
}

@media (min-width:768px) {
  .c2 {
    border-radius: 0 0 0.2rem 0.2rem;
    padding: 4.0rem 0;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 1.8rem;
  }
}

<main
  class="c0"
>
  <div
    aria-label="Nemo enim ipsam"
    class="c1"
    role="img"
    src="https://source.unsplash.com/user/fromitaly/1042x580"
  />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <h2
        class="c4"
      >
        Nemo enim ipsam
      </h2>
      <h3
        class="c5"
      >
        <p>
          Ut labore et 
          <strong>
            dolore magnam
          </strong>
           aliquam
        </p>
      </h3>
      <a
        class="c6"
        href="/properties/nemo-enim-ipsam"
      >
        <span>
          Buy now
        </span>
      </a>
    </div>
  </div>
</main>
`
